staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -58,6 +66,20 @@ func getEventKey(event *v1.Event) string {
 		"")
 }
 
+// getSpamKey builds unique event key based on source, involvedObject
+func getSpamKey(event *v1.Event) string {
+	return strings.Join([]string{
```

Comment:
will leave as-is for now, we do something similar with event aggregation as well, and it has yet to appear on problem list.